test/test_xla_graph_execution.py

```diff
@@ -0,0 +1,74 @@
+# Parse local options first, and rewrite the sys.argv[].
+# We need to do that before import "common", as otherwise we get an error for
+# unrecognized arguments.
+import argparse
+import os
+import sys
+import torch
+import torch_xla
+import torch_xla.core.xla_model as xm
+import torch_xla.utils.utils as xu
+import unittest
+import test_utils
+import time
+
+parser = argparse.ArgumentParser(add_help=False)
+parser.add_argument('--verbosity', type=int, default=0)
+FLAGS, leftovers = parser.parse_known_args()
+sys.argv = [sys.argv[0]] + leftovers
+print(FLAGS)
+
+XLA_DISABLE_FUNCTIONALIZATION = bool(
+    os.environ.get('XLA_DISABLE_FUNCTIONALIZATION', False))
+
+
+class TestXlaGraphExecution(test_utils.XlaTestCase):
+
+  def test_graph_execution_disabled(self):
+    # Test xla_debug_level = 0
+    print("Test xla debug level disabled.")
+    torch_xla._XLAC._set_allow_execution(True)
+    os.environ['PT_XLA_DEBUG_LEVEL'] = '0'
+    start_time = time.time()
+    x = torch.ones(2, device=xm.xla_device())
+    self.assertEqual(x[0], 1.0)  # This should trigger the checking
+
+    print("--- %s seconds ---" % (time.time() - start_time))
+    del x
+
+  def test_graph_execution_warning(self):
+    # Test xla_debug_level = 2 (with WARNING)
+    print("Test xla debug level enabled.")
+    torch_xla._XLAC._set_allow_execution(True)
+    os.environ['PT_XLA_DEBUG_LEVEL'] = '2'
+    start_time = time.time()
+    x = torch.ones(2, device=xm.xla_device())
+    self.assertEqual(x[0], 1.0)  # This should trigger the checking
+    print("--- %s seconds ---" % (time.time() - start_time))
+    del x
+
+  def test_graph_execution_error(self):
+    # Test ERROR level
+    print(
+        "Test check level as runtime error with warning messages before that.")
+    torch_xla._XLAC._set_allow_execution(
+        False)  # this flag disallows graph execution
+    start_time = time.time()
+    x = torch.ones(2, device=xm.xla_device())
+    with self.assertRaises(RuntimeError) as e:
+      self.assertEqual(x[0], 1.0)  # This should trigger the checking
+    print("--- %s seconds ---" % (time.time() - start_time))
+    del x
+    print(
```

Comment:
nit: Any particular reason to add time information to tests? Unless these tests are needed for benchmarking, it is cleaner to not add unnecessary information.